src/Components/Components/src/Rendering/ComponentState.cs

```diff
@@ -236,6 +236,9 @@ public Task DisposeInBatchAsync(RenderBatchBuilder batchBuilder)
                 var result = ((IAsyncDisposable)Component).DisposeAsync();
                 if (result.IsCompletedSuccessfully)
                 {
+                    // If it's a IValueTaskSource backed ValueTask,
+                    // inform it its result has been read so it can reset
+                    result.GetAwaiter().GetResult();
```

Comment:
@davidfowl do we need to do this across all places where we consume ValueTask?

Reply:
Yes. There's an uber issue with a bunch of violations in our code base wrt value task. @stephentoub filed the issue in 5.0 we should address those things in 6.0

Reply:
Looks like it's #16876.  Might as well go ahead and take this if someone wants to sign off on it.